File: seed/php-model/trace/src/Submission/RecordedResponseNotification.php
<?php

namespace Seed\Submission;

use Seed\Core\SerializableType;
use Seed\Core\JsonProperty;

class RecordedResponseNotification extends SerializableType
{
    /**
     * @var string $submissionId
     */
    #[JsonProperty("submissionId")]
    public string $submissionId;

    /**
     * @var int $traceResponsesSize
     */
    #[JsonProperty("traceResponsesSize")]
    public int $traceResponsesSize;

    /**
     * @var ?string $testCaseId
     */
    #[JsonProperty("testCaseId")]
    public ?string $testCaseId;

    /**
     * @param string $submissionId
     * @param int $traceResponsesSize
     * @param ?string $testCaseId
     */
    public function __construct(
        string $submissionId,
        int $traceResponsesSize,
        ?string $testCaseId = null,
    ) {
        $this->submissionId = $submissionId;
        $this->traceResponsesSize = $traceResponsesSize;
        $this->testCaseId = $testCaseId;
    }
}

File: seed/ruby-sdk/trace/lib/seed/submission/types/recorded_response_notification.rb
# frozen_string_literal: true

module Seed
  module Submission
    module Types
      class RecordedResponseNotification < Internal::Types::Model
        field :submission_id, -> { String }, optional: false, nullable: false, api_name: "submissionId"
        field :trace_responses_size, -> { Integer }, optional: false, nullable: false, api_name: "traceResponsesSize"
        field :test_case_id, -> { String }, optional: true, nullable: false, api_name: "testCaseId"
      end
    end
  end
end
